modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -409,10 +416,55 @@ private void onEvent(CoreEvent event, ExecutorCallback executorCallback) {
       };
 
       if (location != null) {
-        ((DefaultFlowCallStack) event.getFlowCallStack())
-            .setCurrentProcessorPath(resolvedProcessorRepresentation);
+        ((DefaultFlowCallStack) event.getFlowCallStack()).setCurrentProcessorPath(resolvedProcessorRepresentation);
+
+        Function<CoreEvent, CoreEvent> resultMapper = isTargetWithPolicies(event) ? (policyResult) -> {
+          if (returnDelegate instanceof PayloadTargetReturnDelegate) {
```

Comment:
This is a no-go. making an instanceof at the return delegate completely defeats its purpose. Also, this code is super fragile and will potentially break each time we add another return delegate. Also, this if is kind of repeating the return delegate business logic to begin with.

Finally, creating lambda functions which carry contextual state is a performance problem. We need to avoid that